### PR TITLE
[#145567117] - add promo code field into registration form

### PR DESCRIPTION
**Pull request by kravchina**

### CODE REVIEW

File: src/partials/registration.html [hunk 1]
```diff
@@ -120,8 +128,8 @@
                             </div>
                         </div>
                         <div class="button-block" style="padding-top: 20px;">
-                            <button class="btn btn-green btn-lg" ng-click="register(practice, invitation)" ng-hide="promo">Register</button>
-                            <button class="btn btn-green btn-lg" ng-click="createPracticeAndRegister(practice, invitation)" ng-show="promo" ng-disabled="form.$invalid || invitation.password_confirmation !== invitation.password || isResend">Register</button>
+                            <button class="btn btn-green btn-lg" ng-click="register(practice, invitation)" ng-hide="promo || promo === ''">Register1</button>
```
Comment (anton-kaidalov) [inline src/partials/registration.html]: @kravchina Correct button captions.

File: src/js/controllers/registrationController.js [hunk 2]
```diff
@@ -86,27 +86,39 @@ angular.module('registration')
...
+
+                                });
+                        }, function(response){
+                            if(response.status === 404) {
+                                Notification.error('promotion.not.found');
+                            }
+                            if(response.status === 422) {
+                                Notification.error('promotion.expired');
+                            }
+                            $scope.form.promo.$invalid = true;
```
[the remark this answers is not in the document]
Comment (kravchina) [reply src/js/controllers/registrationController.js]: No, you are not right. In this case we are handle the specified cases and show specified error messages (Notification.error).
For others statuses we have a special Interceptor and handlers in `app.js`.

[the remark this answers is not in the document]
Comment (anton-kaidalov) [reply src/js/controllers/registrationController.js]: Have you checked that what you are saying is really happening? :)

I'm getting no messages to the user if promo validation fails (e.g. with HTTP 500) in both above-mentioned cases:
* when user navigates to `/register/promo/:promo`;
* when user clicks Register.

We can have a chat if you think we're not on the same page here.

[the remark this answers is not in the document]
Comment (anton-kaidalov) [reply src/js/controllers/registrationController.js]: Discussed with Slava and Anton M in the chat, this problem is out of scope of this PR.

Closing PR.

**Comment by anton-kaidalov** [comment]
UPDATE:

DON'T MERGE THIS PR. It breaks registration by invitation.

**Comment by anton-kaidalov** [comment]
Reviewed. Now we can merge it.

**Pull request merged**